project/Settings.scala

```diff
@@ -0,0 +1,39 @@
+import sbt._
+import org.scalajs.sbtplugin.ScalaJSPlugin.autoImport._
+
+/**
+ * Application settings. Configure the build for your application here.
+ * You normally don't have to touch the actual build definition after this.
+ */
+object Settings {
+  /** The name of your application */
+  val name = "anduin-fsm"
+
+  /** The version of your application */
+  val version = "1.0"
+
+  /** Options for the scala compiler */
+  val scalacOptions = Seq(
+    "-Xlint",
+    "-unchecked",
+    "-deprecation",
+    "-feature"
+  )
```

Comment:
You could make it stricter. Refer [https://tpolecat.github.io/2017/04/25/scalac-flags.html](https://tpolecat.github.io/2017/04/25/scalac-flags.html) for more details.